components/Footer/FooterComponent.vue

```diff
@@ -58,25 +58,129 @@
 </template>
 
 <script setup lang="ts">
-import {
-  FOOTER_ELEMENTS,
-  SECONDARY_FOOTER_LINKS,
-  STAY_CONNECTED_LINKS,
-} from "~/constants/menuLinks";
+import { SocialMedia } from "~/constants/links";
+import type { Link, LinkWithIcon } from "~/types/links";
 
 interface Props {
   oldContainer?: boolean;
   theme?: "light" | "dark";
 }
 
+interface LinksGroup {
+  title: string;
+  elements: Link[];
+}
+
+interface LinksWithIconGroup {
+  title: string;
+  elements: LinkWithIcon[];
+}
+
 withDefaults(defineProps<Props>(), {
   oldContainer: false,
   theme: "light",
 });
 
-const footerElements = FOOTER_ELEMENTS;
-const stayConnectedElements = STAY_CONNECTED_LINKS;
-const secondaryFooterLinks = SECONDARY_FOOTER_LINKS;
+const footerElements: LinksGroup[] = [
+  {
+    title: "Learn",
+    elements: [
+      {
+        label: "Textbook",
+        url: "/learn",
+      },
+      {
+        label: "Tutorials",
+        url: "https://qiskit.org/documentation/tutorials.html",
+        segment: {
+          cta: "tutorials",
+          location: "menu",
+        },
+      },
+      {
+        label: "Videos",
+        url: "https://www.youtube.com/qiskit",
+      },
+    ],
+  },
+  {
+    title: "Community",
+    elements: [
+      {
+        label: "Events",
+        url: "/events",
+        segment: {
+          cta: "events",
+          location: "menu",
+        },
+      },
+      {
+        label: "Advocates",
+        url: "/advocates#become-an-advocate",
+      },
+      {
+        label: "Code of conduct",
+        url: "https://github.com/Qiskit/qiskit/blob/master/CODE_OF_CONDUCT.md",
+        segment: {
+          cta: "code-of-conduct",
+          location: "footer",
+        },
+      },
+    ],
+  },
+  {
+    title: "Support",
+    elements: [
+      SocialMedia.github,
+      SocialMedia.support,
+      {
+        label: "Documentation",
+        url: "https://qiskit.org/documentation/",
+      },
+    ],
+  },
+];
+
+const stayConnectedElements: LinksWithIconGroup = {
+  title: "Stay Connected",
+  elements: [
+    {
+      ...SocialMedia.twitter,
+      icon: LogoTwitter20,
```

Comment:
Heads up: I'm getting this 500 error n the preview branch, and locally as well. (Seems to be happening for all of these `Logo<Icon>20` icons


![Screenshot 2023-05-24 at 9 11 01 AM](https://github.com/Qiskit/qiskit.org/assets/6276074/6468b2cd-e692-4b93-a9dd-571bb7eee777)

Reply:
Works nicely, thanks @eddybrando